hs_core/management/utils.py

```diff
@@ -319,7 +319,9 @@ def check_irods_files(resource, stop_on_error=False, log_errors=True,
                 msg = "check_irods_files: django file {} does not exist in iRODS"\
                     .format(file_storage_path)
                 if clean_django and not dry_run:
-                    delete_resource_file(resource.short_id, f.short_path, resource.creator,
+                    if not user:
+                        user = resource.creator
+                    delete_resource_file(resource.short_id, f.short_path, user,
```

Comment:
Is this going to raise `ValidationError` if a non-admin user is deleting a file of a published resource? 
https://github.com/hydroshare/hydroshare/blob/develop/hs_core/hydroshare/resource.py#L916

Reply:
yes, that had been the default behavior of the `repair_resource` management command (if you ran it with `--publish` flag and it actually encountered file issue, it would fail with ValidationError).

But this PR adds `--admin` flag so that the published resources actually get repaired.

I suppose we could handle the ValidationError more gracefully, but the stack trace works fine when running with Jenkins so I'm inclined to leave it unless I'm missing your point